File: Barchart.BinarySerializer/Buffers/Factories/DataBufferReaderFactory.cs
namespace Barchart.BinarySerializer.Buffers.Factories;

/// <summary>
///     An implementation of <see cref="IDataBufferReaderFactory" /> that uses byte arrays 
///     as the underlying binary data storage foreach <see cref="IDataBufferReader" />.
/// </summary>
public class DataBufferReaderFactory : IDataBufferReaderFactory
{
    #region Methods

    /// <inheritdoc />
    public IDataBufferReader Make(byte[] byteArray)
    {
        return new DataBufferReader(byteArray);
    }
    
    #endregion
}

File: typescript/src/buffers/factories/data-buffer-reader-factory.ts
import { DataBufferReader } from "../data-buffer-reader";
import { DataReader } from "../data-reader.interface";
import { DataReaderFactory } from "./data-reader-factory.interface";

/**
 * An implementation of DataReaderFactory that uses
 * byte arrays as the underlying binary data storage for
 * each DataReader.
 *
 * @public
 * @exported
 * @implements {DataReaderFactory}
 */
export class DataBufferReaderFactory implements DataReaderFactory {
    make(bytes: Uint8Array): DataReader {
        return new DataBufferReader(bytes);
    }
}